qiskit/transpiler/passes/optimization/commutative_inverse_cancellation.py

```diff
@@ -0,0 +1,91 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Cancel pairs of inverse gates exploiting commutation relations."""
+
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class CommutativeInverseCancellation(TransformationPass):
+    """Cancel pairs of inverse gates exploiting commutation relations."""
+
+    def _skip_node(self, node):
+        """Returns True if we should skip this node for the analysis."""
+        if node.op._directive or node.name in {"measure", "reset", "delay"}:
+            return True
+        if node.op.condition:
+            return True
+        # ToDo: Not sure about the next line
+        if node.op.is_parameterized():
+            return True
+        # ToDo: Even less sure about the next line
+        # if node.op.params:
+        #     return True
```

Comment:
I think the `is_parameterized()` check should cover this too.

Reply:
Oh, I see the difference now: `gate.is_parameterized()` returns `True` for a parameterized gate with an unbound parameter (such as `RZGate(Parameter("Theta"))`), but returns `False` for a parameterized gate with a bound parameter (such as `RZGate(np.pi/2)`). The current analysis takes the over-conservative approach in the unbound case (skipping such gates or declaring that such gates do not commute with other gates), and works well in the bound case (for instance, cancelling the inverse pair of `RZGate(np.pi/2)` and `RZGate(-np.pi/2)`).